atmo/jobs/models.py

```diff
@@ -120,13 +112,15 @@ def has_never_run(self):
         Looks at both the cluster status and our own record when
         we asked it to run.
         """
-        return (self.most_recent_status == self.DEFAULT_STATUS or
-                self.last_run_date is None)
+        return (self.latest_run is None or
+                self.latest_run.status == DEFAULT_STATUS or
+                self.latest_run.run_date is None)
```

Comment:
This may be easier to say `self.runs.exclude(status=DEFAULT_STATUS).exists()`? I now wonder if we should have a spark job run *always* have a non-empty status. Even if it's "CREATED" only, so we can filter them out.